folio-spring-i18n/src/main/java/org/folio/spring/i18n/service/TranslationService.java

```diff
@@ -0,0 +1,352 @@
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.spring.i18n.config.TranslationConfiguration;
+import org.folio.spring.i18n.model.LanguageRegionPair;
+import org.folio.spring.i18n.model.TranslationFile;
+import org.folio.spring.i18n.model.TranslationMap;
+import org.folio.spring.i18n.model.TranslationMatchQuality;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.ResourcePatternResolver;
+import org.springframework.stereotype.Service;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+/**
+ * A {@link Service Service} to manage translations as a whole.  In reality, the most important
+ * method is simply the format method which will do all of the heavy lifting as needed.  This
+ * service will keep track of translation files and locales, caching as possible, in order to
+ * serve the user the best possible translation.
+ */
+@Log4j2
+@Service
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class TranslationService {
+
+  /**
+   * The location of all translations in the classpath. Allows for format matching UI modules:
+   *  /translations/{module-name}/{locale}.json
+   */
+  private static final String TRANSLATIONS_CLASSPATH = "classpath:%s*/*.json";
+
+  /**
+   * A map from language code to a map from country to the JSON resource containing {key to ICU format string mappings}.
+   * Will be populated the first time any translation is requested.
+   * Note that files are not read here, only enumerated.  Look at the other maps for that information
+   *
+   * <p>This can be null - if this is the case, the map has not been constructed yet.</p>
+   * <p>You should call getFileMap instead of accessing this directly.</p>
+   *
+   * <p>
+   *  Language code ({@link TranslationFile#UNKNOWN_PART TranslationFile#UNKNOWN_PART}) will be used if unknown
+   * </p>
+   */
+  @Nullable
+  protected Map<String, Map<String, TranslationFile>> translationFileFromLanguageCountryMap =
+    null;
+
+  /**
+   * A map from locales to translations, filled in on-demand as locales are presented.
+   */
+  protected Map<Locale, TranslationMap> localeTranslations = new HashMap<>();
+
+  private final ResourcePatternResolver resourceResolver;
+  private final TranslationConfiguration configuration;
+
+  /**
+   * Get a map from language -&gt; country -&gt; {@link TranslationFile TranslationFile}.
+   * The files are pulled from the filesystem based on the provided
+   * {@link TranslationConfiguration TranslationConfiguration}.
+   *
+   * @return the map of languages to countries to {@link TranslationFile TranslationFile}s
+   */
+  @Nonnull
+  public Map<String, Map<String, TranslationFile>> getFileMap() {
```

Comment:
Why do we need `getFileMap`, `getTranslation`, `getTranslation`, `getCurrentLocales` and so on to be public? I see in documentation usecases only for `format` and `formatList`. And method with name getTranslation is confusing without reading javadoc it's not clear why it not translate anything.